### PR TITLE
Add config to customise transport ip port, storage path and accessory pin

### PR DESCRIPTION
**Pull request by brutella**
Similar to #35 and #37

### CODE REVIEW

File: _example/example.go [hunk 1]
```diff
@@ -14,7 +14,9 @@ func main() {
 		Name: "Lamp",
 	}
 	sw := accessory.NewSwitch(switchInfo)
-	t, err := hap.NewIPTransport("00102003", sw.Accessory)
+
+	config := hap.Config{Pin: "12344321", Port: ":12345", StoragePath: "db"}
```
Comment (bmurray) [inline _example/example.go]: Should the port maybe not include the colon before the numbers?   ":12345" should maybe be "12345", or even an integer, 12345?

**Comment by bmurray** [comment]
It does break backwards compatibility, but this is definitely the right solution.

File: _example/example.go (hunk 1 above)
Comment (brutella) [inline _example/example.go]: Good point

**Comment by brutella** [comment]
@bmurray At this point, backwards compatibility is not the highest priority

**Pull request merged**